src/main/java/org/apache/mesos/scheduler/plan/DefaultBlockFactory.java

```diff
@@ -28,32 +28,46 @@ public DefaultBlockFactory(StateStore stateStore) {
     }
 
     @Override
-    public Block getBlock(String taskType, TaskSpecification taskSpecification) throws InvalidRequirementException {
+    public Block getBlock(TaskSpecification taskSpecification) throws Block.InvalidException {
         logger.info("Generating block for: " + taskSpecification.getName());
         Optional<Protos.TaskInfo> taskInfoOptional = stateStore.fetchTask(taskSpecification.getName());
-        if (!taskInfoOptional.isPresent()) {
-            logger.info("Generating new block for: " + taskSpecification.getName());
-            return new DefaultBlock(
-                    taskSpecification.getName(),
-                    offerRequirementProvider.getNewOfferRequirement(taskType, taskSpecification),
-                    Status.PENDING);
-        } else {
-            try {
+
+        try {
+            if (!taskInfoOptional.isPresent()) {
+                logger.info("Generating new block for: " + taskSpecification.getName());
```

Comment:
Redundant in what sense?  There are four log lines.

1. Trying
2. New
3. Existing
4. Failed